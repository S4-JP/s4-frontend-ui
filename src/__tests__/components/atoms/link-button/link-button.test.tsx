import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import { LinkButton, Color } from '../../../../index'

/**
 * type Props = {
 *   href?: string
 *   title: string
 *   date?: string
 *   disabled?: boolean
 * }
 */
describe('link-button のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(<LinkButton title="test" />)

    expect(getByTestId('atoms-lb-wrap')).toMatchSnapshot()
  })

  test('props.hrefを設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(<LinkButton title="test" href="/test" />)
    expect(getByTestId('atoms-lb-wrap')).toHaveStyle(
      `background-color: ${Color.TEXT.GRAY};`
    )

    fireEvent.mouseOver(getByTestId('atoms-lb-wrap'))
    waitFor(() =>
      expect(getByTestId('atoms-lb-wrap')).toHaveStyle(
        `background: ${Color.COMPONENT.GRAY_HOVER};`
      )
    )
  })

  test('props.handleClick設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(
      <LinkButton title="test" handleClick={jest.fn()} />
    )

    expect(getByTestId('atoms-lb-icon-wrapper')).toMatchSnapshot()
  })
})
