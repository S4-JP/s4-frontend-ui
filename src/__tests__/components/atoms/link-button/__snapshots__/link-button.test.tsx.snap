// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link-button のレンダリング確認 props.handleClick設定した場合、設定値でレンダリングされる。 1`] = `
<div
  class="sc-gFGZVQ bGdXuU"
  data-testid="atoms-lb-icon-wrapper"
>
  <div
    class="sc-lbOyJj hNMqcY"
    data-testid="atoms-lb-button-title"
  >
    test
    <div
      class="sc-ckMVTt eyDyOf"
      data-testid="atoms-lb-date-wrap"
    >
      <span
        class="sc-papXJ jTrzrE"
        color="#222E40"
        data-testid="atoms-i-icon-wrap"
        size="11"
      >
        <svg
          data-testid="atoms-icon-svg-history"
          viewBox="0 0 40 40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M40 40H0V0h40z"
            fill="none"
          />
          <path
            d="m38.9 7.864-.954-.412-.273-.051a2.7 2.7 0 0 0-2.807 1.874c-.081.236-.172.465-.26.7l-.159.415a18.436 18.436 0 0 0-7.83-7.116l-.045-.1-2.581-.97a34.77 34.77 0 0 1-.92-.212 17.39 17.39 0 0 0-1.587-.332A18.367 18.367 0 0 0 6.348 6.033a18.048 18.048 0 0 0-4.585 6.064l-.934 2.342-.1.476a18 18 0 0 0 .509 11.649A18.02 18.02 0 0 0 11.97 37.3q.711.268 1.463.493a17.708 17.708 0 0 0 10.638-.1 17.866 17.866 0 0 0 10.041-7.737 2.87 2.87 0 0 0 .234-.474 2.556 2.556 0 0 0-1.481-3.343c-.854-.323-2.127-.342-3.341 1.4a13.307 13.307 0 1 1-8.761-20.65 12.8 12.8 0 0 1 9.511 6.843l-2.886-1.093a2.641 2.641 0 0 0-3.51 1.466 2.565 2.565 0 0 0 .047 1.98 2.8 2.8 0 0 0 1.6 1.456l7.775 2.933a2.622 2.622 0 0 0 3.562-1.622l2.918-7.742a3.676 3.676 0 0 0 .188-.666 2.732 2.732 0 0 0-1.068-2.58"
          />
          <path
            d="M17.659 9.917a2.181 2.181 0 0 0-2.182 2.182v10.8l9.215 3.5a2.182 2.182 0 1 0 1.548-4.08l-6.403-2.43v-7.79a2.181 2.181 0 0 0-2.182-2.182"
          />
        </svg>
      </span>
      <span
        class="sc-fbPSWO hXOTvw"
        data-testid="atoms-lb-date-text"
      />
    </div>
  </div>
</div>
`;

exports[`link-button のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-ehmTmK hPaqzJ"
  data-testid="atoms-lb-wrap"
  title="test"
>
  <div
    class="sc-gFGZVQ bGdXuU"
    data-testid="atoms-lb-icon-wrapper"
  >
    <div
      class="sc-lbOyJj hNMqcY"
      data-testid="atoms-lb-button-title"
    >
      test
      <div
        class="sc-ckMVTt eyDyOf"
        data-testid="atoms-lb-date-wrap"
      >
        <span
          class="sc-papXJ kLvGuc"
          data-testid="atoms-i-icon-wrap"
          size="11"
        >
          <svg
            data-testid="atoms-icon-svg-history"
            viewBox="0 0 40 40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M40 40H0V0h40z"
              fill="none"
            />
            <path
              d="m38.9 7.864-.954-.412-.273-.051a2.7 2.7 0 0 0-2.807 1.874c-.081.236-.172.465-.26.7l-.159.415a18.436 18.436 0 0 0-7.83-7.116l-.045-.1-2.581-.97a34.77 34.77 0 0 1-.92-.212 17.39 17.39 0 0 0-1.587-.332A18.367 18.367 0 0 0 6.348 6.033a18.048 18.048 0 0 0-4.585 6.064l-.934 2.342-.1.476a18 18 0 0 0 .509 11.649A18.02 18.02 0 0 0 11.97 37.3q.711.268 1.463.493a17.708 17.708 0 0 0 10.638-.1 17.866 17.866 0 0 0 10.041-7.737 2.87 2.87 0 0 0 .234-.474 2.556 2.556 0 0 0-1.481-3.343c-.854-.323-2.127-.342-3.341 1.4a13.307 13.307 0 1 1-8.761-20.65 12.8 12.8 0 0 1 9.511 6.843l-2.886-1.093a2.641 2.641 0 0 0-3.51 1.466 2.565 2.565 0 0 0 .047 1.98 2.8 2.8 0 0 0 1.6 1.456l7.775 2.933a2.622 2.622 0 0 0 3.562-1.622l2.918-7.742a3.676 3.676 0 0 0 .188-.666 2.732 2.732 0 0 0-1.068-2.58"
            />
            <path
              d="M17.659 9.917a2.181 2.181 0 0 0-2.182 2.182v10.8l9.215 3.5a2.182 2.182 0 1 0 1.548-4.08l-6.403-2.43v-7.79a2.181 2.181 0 0 0-2.182-2.182"
            />
          </svg>
        </span>
        <span
          class="sc-fbPSWO hXOTvw"
          data-testid="atoms-lb-date-text"
        />
      </div>
    </div>
    <span
      class="sc-papXJ cBqjuY"
      data-testid="atoms-i-icon-wrap"
    >
      <svg
        data-testid="atoms-icon-svg-external-link"
        viewBox="0 0 40 40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h40v40H0z"
          fill="none"
        />
        <path
          d="M37.135 24.251a2.806 2.806 0 0 0-2.8 2.8v4.36a2.918 2.918 0 0 1-2.914 2.916H8.524a2.919 2.919 0 0 1-2.916-2.916V8.524a2.919 2.919 0 0 1 2.916-2.916h4.731a2.804 2.804 0 0 0 0-5.608H8.524A8.533 8.533 0 0 0 0 8.524v22.892a8.532 8.532 0 0 0 8.524 8.524h22.892a8.532 8.532 0 0 0 8.522-8.524v-4.36a2.8 2.8 0 0 0-2.803-2.805Z"
        />
        <path
          d="M39.143 1.382A2.911 2.911 0 0 0 37.07.524H24.296a2.927 2.927 0 0 0-2.924 2.931 2.839 2.839 0 0 0 .035.452 2.994 2.994 0 0 0 2.976 2.412l5.638.01-10.216 10.225a2.984 2.984 0 0 0-.422 3.9 2.93 2.93 0 0 0 4.088.662 3.076 3.076 0 0 0 .375-.321l10.322-10.322v5.608a2.994 2.994 0 0 0 2.457 3.011 2.927 2.927 0 0 0 3.341-2.442 3.37 3.37 0 0 0 .034-.417V3.454a2.909 2.909 0 0 0-.857-2.072Z"
        />
      </svg>
    </span>
  </div>
</div>
`;
